src/cluster_legacy.c

```diff
@@ -3023,6 +3107,19 @@ writeSdsPingExtIfNonempty(uint32_t *totlen_ptr, clusterMsgPingExt **cursor_ptr,
     return 1;
 }
 
+static uint32_t
+writePortPingExtIfNonzero(uint32_t *totlen_ptr, clusterMsgPingExt **cursor_ptr, clusterMsgPingtypes type, uint16_t value) {
+    if (value == 0) return 0;
+    size_t size = getAlignedPingExtSize(sizeof(clusterMsgPingExtClientPort));
+    if (*cursor_ptr != NULL) {
+        void *ext = preparePingExt(*cursor_ptr, type, size);
+        memcpy(ext, &value, sizeof(value));
```

Comment:
We need to handle endianess. All numbers in the cluster bus are stored in network byte order (big endian). Let's convert value before memcpy?

```suggestion
        value = htons(value);
        memcpy(ext, &value, sizeof(value));
```

Reply:
Good point, thanks! I've applied all your suggestions in a single commit, as GitHub wouldn't let me add them to a batch for some reason.

Reply:
Great. In the diff view, you can add them to a batch, but not in the "Conversation" view.